apps/frontend/src/app/5_pages/AavePage/components/LendPositionsList/components/WithdrawForm/WithdrawForm.tsx

```diff
@@ -0,0 +1,150 @@
+import { Decimal } from '@sovryn/utils';
+
+import { BOB_CHAIN_ID } from '../../../../../../../config/chains';
+
+import { AmountRenderer } from '../../../../../../2_molecules/AmountRenderer/AmountRenderer';
+import { AssetAmountInput } from '../../../../../../2_molecules/AssetAmountInput/AssetAmountInput';
+import { AssetRenderer } from '../../../../../../2_molecules/AssetRenderer/AssetRenderer';
+import { useAaveUserReservesData } from '../../../../../../../hooks/aave/useAaveUserReservesData';
+import { useAaveWithdraw } from '../../../../../../../hooks/aave/useAaveWithdraw';
+import { useDecimalAmountInput } from '../../../../../../../hooks/useDecimalAmountInput';
+import { translations } from '../../../../../../../locales/i18n';
+import { tabItems } from './WithdrawForm.constants';
+
+const pageTranslations = translations.aavePage;
+
+type WithdrawFormProps = {
+  asset: string;
+  onComplete: () => void;
+};
+
+export const WithdrawForm: FC<WithdrawFormProps> = ({ asset, onComplete }) => {
+  const { handleWithdraw } = useAaveWithdraw();
+  const { summary } = useAaveUserReservesData();
+  const [withdrawAsset, setWithdrawAsset] = useState(asset);
+  const [withdrawAmount, setWithdrawAmount, withdrawSize] =
+    useDecimalAmountInput('');
+
+  const withdrawableAssetsOptions = useMemo(
+    () =>
+      summary.reserves
+        .filter(r => r.supplied.gt(0))
+        .map(sa => ({
+          value: sa.asset,
+          label: (
+            <AssetRenderer
+              showAssetLogo
+              asset={sa.asset}
+              assetClassName="font-medium"
+              chainId={BOB_CHAIN_ID}
+            />
+          ),
+        })),
+    [summary],
+  );
+
+  const withdrawReserve = useMemo(() => {
+    return summary.reserves.find(r => r.reserve.symbol === withdrawAsset);
+  }, [withdrawAsset, summary]);
+
+  const maximumWithdrawAmount: Decimal = useMemo(() => {
+    return withdrawReserve ? withdrawReserve.supplied : Decimal.from(0);
+  }, [withdrawReserve]);
+
+  const withdrawAmountUsd: Decimal = useMemo(() => {
+    return withdrawReserve
+      ? withdrawSize.mul(withdrawReserve.reserve.priceInUSD)
+      : Decimal.from(0);
+  }, [withdrawSize, withdrawReserve]);
+
+  const remainingSupply = useMemo(
+    () => maximumWithdrawAmount.sub(withdrawSize),
+    [withdrawSize, maximumWithdrawAmount],
+  );
+
+  const isValidWithdrawAmount = useMemo(
+    () => (withdrawSize.gt(0) ? withdrawSize.lte(maximumWithdrawAmount) : true),
+    [withdrawSize, maximumWithdrawAmount],
+  );
+
+  const submitButtonDisabled = useMemo(
+    () => !isValidWithdrawAmount || withdrawSize.lte(0),
+    [isValidWithdrawAmount, withdrawSize],
+  );
+
+  const onConfirm = useCallback(() => {
+    handleWithdraw(
+      withdrawSize,
+      withdrawAsset,
+      withdrawSize.eq(maximumWithdrawAmount),
+      { onComplete },
+    );
+  }, [
+    handleWithdraw,
+    withdrawSize,
+    withdrawAsset,
+    onComplete,
+    maximumWithdrawAmount,
+  ]);
+
```

Comment:
Same here with the onComplete. Just to compare both solutions

Reply:
Same, we don't have access to a promise